.circleci/config.yml

```diff
@@ -818,182 +818,52 @@ jobs:
             npm --version
             test/externalTests/solc-js/solc-js.sh /tmp/workspace/soljson.js $(cat /tmp/workspace/version.txt)
 
-  t_ems_compile_ext_gnosis:
+  t_ems_compile_ext:
+    parameters:
+      project:
+        type: string
+      nodejs_version:
+        type: integer
+        default: 14
     docker:
-      - image: circleci/node:14
-    environment:
-      TERM: xterm
-    steps:
-      - checkout
-      - attach_workspace:
-          at: /tmp/workspace
-      - run:
-          name: External GnosisSafe compilation
-          command: |
-            export COMPILE_ONLY=1
-            test/externalTests/gnosis.sh /tmp/workspace/soljson.js
-
-  t_ems_test_ext_gnosis:
-    docker:
-      # NOTE: Tests do not start on node.js 14 ("ganache-cli exited early with code 1").
-      - image: circleci/node:12
-    environment:
-      TERM: xterm
-    steps:
-      - checkout
-      - attach_workspace:
-          at: /tmp/workspace
-      - run:
-          name: External GnosisSafe tests
-          command: |
-            test/externalTests/gnosis.sh /tmp/workspace/soljson.js
-      # TODO: Re-enable notifications once this is running nightly rather than as a PR check
-      #- run: *gitter_notify_failure
-      #- run: *gitter_notify_success
-
-  t_ems_compile_ext_gnosis_v2:
-    docker:
-      - image: circleci/node:14
-    environment:
-      TERM: xterm
-    steps:
-      - checkout
-      - attach_workspace:
-          at: /tmp/workspace
-      - run:
-          name: External GnosisSafe v2 compilation
-          command: |
-            export COMPILE_ONLY=1
-            test/externalTests/gnosis-v2.sh /tmp/workspace/soljson.js
-
-  t_ems_test_ext_gnosis_v2:
-    docker:
-      # NOTE: Tests do not start on node.js 14 ("ganache-cli exited early with code 1").
-      - image: circleci/node:12
-    environment:
-      TERM: xterm
-    steps:
-      - checkout
-      - attach_workspace:
-          at: /tmp/workspace
-      - run:
-          name: External GnosisSafe v2 tests
-          command: |
-            test/externalTests/gnosis-v2.sh /tmp/workspace/soljson.js
-      # TODO: Re-enable notifications once this is running nightly rather than as a PR check
-      #- run: *gitter_notify_failure
-      #- run: *gitter_notify_success
-
-  t_ems_compile_ext_zeppelin:
-    docker:
-      - image: circleci/node:14
-    environment:
-      TERM: xterm
-    steps:
-      - checkout
-      - attach_workspace:
-          at: /tmp/workspace
-      - run:
-          name: External Zeppelin compilation
-          command: |
-            export COMPILE_ONLY=1
-            test/externalTests/zeppelin.sh /tmp/workspace/soljson.js
-
-  t_ems_test_ext_zeppelin:
-    docker:
-      - image: circleci/node:14
+      - image: circleci/node:<<parameters.nodejs_version>>
     environment:
       TERM: xterm
+      COMPILE_ONLY: 1
```

Comment:
Could this also be parameterised so we have a single definition for compile and test and simply pass `1` for compilation only and `0` for test?

Reply:
It could but I originally didn't want to do that because it makes the job list in the workflow longer.

Well, I had to add a parameter for notifications anyway after rebasing on the PR that makes Colony run nigthly so I added one for `compile_only` now too. Check it out.